src/main/java/org/snomed/quality/validator/mrcm/ReportService.java

```diff
@@ -59,78 +57,68 @@ private void createSkippedAssertionsReport(File resultDir, ValidationRun run) th
-				builder.append(failed.getCurrentViolatedConceptIds().size());
-				builder.append(TAB);
-				builder.append(extractInstances(failed.getCurrentViolatedConceptIds()).toString());
-				builder.append(TAB);
-				builder.append(failed.getPreviousViolatedConceptIds().size());
-				builder.append(TAB);
-				builder.append(extractInstances(failed.getPreviousViolatedConceptIds()).toString());
-				builder.append(NEW_LINE);
-				writer.write(builder.toString());
+				writer.write(counter++ + TAB + failed.getUuid().toString() + TAB + failed.getAssertionText() +
+						TAB + failed.getMessage() + TAB + failed.getCurrentViolatedConceptIds().size() + TAB +
+						extractInstances(failed.getCurrentViolatedConceptIds()) + TAB + failed.getPreviousViolatedConceptIds().size() +
+						TAB + extractInstances(failed.getPreviousViolatedConceptIds()) + NEW_LINE);
 			}
 		}
 	}
 
-
 	private void createSummaryReport(File resultDir, String releasePackage, ValidationRun run) throws IOException {
-		File report = new File(resultDir,"MRCMSummaryReport.txt");
-		int totalAttribute = 0;
-		for (String key: run.getMRCMDomains().keySet()) {
-			Domain domain = run.getMRCMDomains().get(key);
-			totalAttribute += domain.getAttributes().size();
-		}
+		final File report = new File(resultDir,"MRCMSummaryReport.txt");
+		final int totalAttribute = run.getMRCMDomains().keySet().stream().map(key -> run.getMRCMDomains().get(key))
+				.mapToInt(domain -> domain.getAttributes().size()).sum();
 
-		try (BufferedWriter writer = new BufferedWriter(new FileWriter(report))) {
-			StringBuilder reportSummary = new StringBuilder();
-			reportSummary.append("ReleasePackage:" + releasePackage);
-			reportSummary.append(NEW_LINE);
-			reportSummary.append("ReportingForReleaseCycle:" + run.getReleaseDate());
-			reportSummary.append(NEW_LINE);
-			reportSummary.append("isStatedViewOnly:" + run.isStatedView());
-			reportSummary.append(NEW_LINE);
-			reportSummary.append("Total MRCM domains loaded:" + run.getMRCMDomains().size());
-			reportSummary.append(NEW_LINE);
-			reportSummary.append("Total MRCM attributes loaded:" + totalAttribute);
-			reportSummary.append(NEW_LINE);
-			reportSummary.append("Total assertions completed:" + run.getCompletedAssertions().size());
-			reportSummary.append(NEW_LINE);
+		try (final BufferedWriter writer = new BufferedWriter(new FileWriter(report, true))) {
+			final StringBuilder reportSummary = new StringBuilder();
+			if (report.length() == 0) {
+				reportSummary.append("ReleasePackage: ")
+						.append(releasePackage)
+						.append(NEW_LINE)
+						.append("ReportingForReleaseCycle: ")
+						.append(run.getReleaseDate())
+						.append(NEW_LINE)
+						.append(NEW_LINE);
+			}
+			reportSummary.append(run.getValidationView() == ValidationView.STATED ? "Results on stated view: " : "Results on inferred view: ")
+					.append(NEW_LINE)
+					.append("\tTotal MRCM domains loaded: ")
+					.append(run.getMRCMDomains().size())
+					.append(NEW_LINE)
```

Comment:
It would be nice extract the messages to the constants class.

Reply:
Fixed.